Various improvements/bug fixes to PP

medariox opened the pull request:
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

ratoaq2 commented on medusa/name_parser/parser.py:
```diff
@@ -359,6 +356,17 @@ def __str__(self):
+        quality = common.Quality.from_guessit(guess)
+        if quality != common.Quality.UNKNOWN:
+            return quality
+        return common.Quality.nameQuality(self.original_name, self.is_anime, extend)
```
Are these properties already initialized when calling this from the init?

medariox replied:
Good question. They should be: https://github.com/pymedusa/Medusa/blob/2697d373701d7bea6dcfe6a651e3d2ff0a23df80/medusa/name_parser/parser.py#L315-L319

self.original_name and self.ab_episode_numbers

medariox commented:
The PR is concluded for me. Test/review/feedback appreciated.

fernandog commented:
1.8 or 1.7 ?

medariox commented:
I didn't change any logic in this PR. I fixed what was supposed to be working, but wasn't and added the quality to ParseResult. That's about it. For me this is a low risk PR can easily go in 1.7. I'm testing it in production, yes.

fernandog commented on medusa/name_parser/parser.py:
```diff
@@ -90,7 +91,7 @@ def _parse_string(self, name):
                 season_number = int(sql_result[0][0])
                 episode_numbers = [int(sql_result[0][1])]
-                logger.debug('Database info for show {name}: S{season} E{episodes}',
+                logger.debug('Database info for show {name}: Season: {season} Episode(s): {episodes}',
```
@ratoaq2 we can have multiple episodes here right?

medariox commented on medusa/name_parser/parser.py:
```diff
@@ -160,15 +161,15 @@ def _parse_string(self, name):
-                        logger.debug('Scene anime show {name} using indexer with absolute {absolute}: S{season}E{episodes}',
-                                     name=result.show.name, absolute=a, season=s, episodes=e)
+                        logger.debug('Scene anime show {name} using indexer with absolute {absolute}: {ep}',
+                                     name=result.show.name, absolute=a, ep=episode_num(s, e))
```
@fernandog ^

fernandog replied:
yes

fernandog commented:
@medariox issue is fixed but i think you should keep guessit (remove last commit)

fernandog commented:
I didn't notice PP more slow with guessit
Did you compare results to decide to not use guessit ?

fernandog commented on medusa/processTV.py:
```diff
@@ -489,54 +487,55 @@ def unRAR(path, rarFiles, force, result):  # pylint: disable=too-many-branches,t
 
-    # Avoid processing the same dir again if we use a process method <> move
+    file_path = os.path.join(dir_name, video_file)
+    file_size = os.path.getsize(file_path)
```
Need to check if file exits here

medariox replied:
Makes the check useless. The whole point is to make sure the file is the same by checking the size.

fernandog commented on medusa/processTV.py:
```diff
@@ -489,54 +487,55 @@ def unRAR(path, rarFiles, force, result):  # pylint: disable=too-many-branches,t
+    # Try generic method first
+    episodes_result = main_db_con.select(
+        'SELECT file_size '
+        'FROM tv_episodes '
```
already processed check should be done in history table not in tv_episode. If user processed multiple releases, you will only check the last one by checking tv_episodes table

medariox replied:
No, for a few reasons:
- History can be deleted
- Download, replace, download same episode would fail
- We want to know what we currently have (that's why the size check) and not what we ever got (history)

fernandog commented on medusa/processTV.py:
```diff
@@ -489,54 +487,55 @@ def unRAR(path, rarFiles, force, result):  # pylint: disable=too-many-branches,t
-        # result.output += logHelper(u"You're trying to post process a video that's already been processed, skipping", logger.DEBUG)
+    history_result = main_db_con.select(
+        'SELECT file_size '
+        'FROM tv_episodes e '
```
already processed check should be done in history table not in tv_episode. If user processed multiple releases, you will only check the last one by checking tv_episodes table

fernandog commented:
approved. more tests in develop
